compat/src/main/scala-2.11_2.12/scala/collection/compat/immutable/ArraySeq.scala

```diff
@@ -115,13 +115,21 @@ object ArraySeq {
         ArrayBuilder.make[T]()(m) mapResult ArraySeq.unsafeWrapArray[T]
     }
 
+  private def unsupportedUpdate: Nothing =
+    throw new UnsupportedOperationException("immutable ArraySeq does not support mutation")
+
+  private final val updateDeprecationMsg =
+    "Mutation is not supported for immutable ArraySeq, and should not have ever been " +
+      "part of the API"
+
   @SerialVersionUID(3L)
   final class ofRef[T <: AnyRef](val unsafeArray: Array[T]) extends ArraySeq[T] with Serializable {
     lazy val elemTag         = ClassTag[T](unsafeArray.getClass.getComponentType)
     def length: Int          = unsafeArray.length
     def apply(index: Int): T = unsafeArray(index)
-    def update(index: Int, elem: T) { unsafeArray(index) = elem }
-    override def hashCode = MurmurHash3.arrayHash(unsafeArray, MurmurHash3.seqSeed)
+    @deprecated(updateDeprecationMsg, since = "2.3.3")
+    def update(index: Int, elem: T): Unit = unsupportedUpdate
```

Comment:
I guess we can assume that nobody has relied on this previous implementation since code calling `update` would not have compiled with Scala 2.13, and the purpose of scala-collection-compat is to cross-compile with 2.12 and 2.13…

So, I think it’s OK to throw an exception here.

Reply:
not only that, but since it's only defined on the implementations (`ofRef`, `ofInt`, etc.) and not `ArraySeq` itself, you would have had to manually create an instance of the implementation type, which is both less convenient and unlikely